### DISCUSSION TITLE
How to tune model parameters

### DISCUSSION THREAD

**Discussion by famy040326** [mindee/doctr#1269] [Q&A] [answered]
I am trying to improve the recognition rate for an image with certain characters by tuning the model.
The characters to be recognized are a random alphabet and a sequence of numbers.

I would like to ask a question because I am not sure what some of the parameters represent in each parameter when training a model.
And I would like to know if you know the correlation between each parameter.
Here are my questions Please answer.

1. I have read the help for --train-samples, --val-samples, --min-chars, and --max-chars, but I would like to know more details about each of them.
How can I use them effectively?

2. I have a question about the correlation of each parameter.
I understand that --epochs, --batch_size, and --lr are generally effective when adjusted.
Are there any other parameters that are correlated?
Are all but three effective on their own?

3. I have a question about how to proceed with tuning.
I would limit myself to --epochs, --batch_size, and --lr and vary the values to check the recognition rate results. Is this the correct way to proceed?
Ideally I would like to tune without limiting the parameters, but I do not have the resources.
If there is a better way to proceed, please let me know.
We have about 2000 images for training with real data.

Thank you in advance for your cooperation in the above.

Translated with www.DeepL.com/Translator (free version)

**Comment by felixT2K** [comment]
Hi @famy040326 👋🏼,

`--train-samples:` the number of generated images for training - image generator (only if you don't provide your own dataset)
`--val-samples:` same as above for validation
`--min-chars / --max-chars :` the minimum number of printed chars in a generated image (does only effect if you use the image generator and not provide your own data)

`-- epochs:` how long the model should train (1 epoch == all data feed trough the network) set it to a high number like 200 if the metrics / loss does not improve you can cancel the run manual
`-- batch_size:` depends on the available GPU VRAM -> how many samples are processed at onces
`-- lr:` for fine tuning 0.0001 should be a good value

2000 images are much to less for training from scratch so i suggest to fine tune the model (set `--pretrained`)

Could you provide some more information about:
 - backend ( TensorFlow or PyTorch ) ?
 - model you want to fine tune ?
 - on which vocab do you want to train ?

**Reply by famy040326** [reply]
Hi @felixT2K,

I will answer your question first.
backend ( TensorFlow or PyTorch ) ?
 -TensorFlow.
model you want to fine tune ?
 -crnn_vgg16_bn.
on which vocab do you want to train ?
 -english.

And thank you for answering my question.
Since I have 2000 pieces of reality data, can I specify both -train-samples and -val-samples as 0 to run?
Or do I need to set something?

**Reply by felixT2K** [reply]
You don't need to specify set both (--train-samples and --val-samples) it will be ignored if you pass a path to your own train and val set
`--pretrained` will in TF only work if you don't specify the `--vocab`currently but that's ok because the default vocab is `legacy_french`which contains all english vocab chars

**Reply by famy040326** [reply]
By setting the train and val paths
I see that --train-samples and --val-samples are ignored.
I also understand that specifying --pretrained will use the legacy_french vocabulary.

So basically I would be adjusting epochs, batch_size, and lr?

**Reply by felixT2K** [reply, marked as answer]
yes :) batch_size according to your hardware

**Reply by famy040326** [reply]
Thank you for responding to my question. It was very helpful.
